### PR TITLE
Bump Xcode version to 13.0.0

### PR DESCRIPTION
**Pull request by OdNairy**
- [x] Update API Digester baseline for Xcode 13 format
- [x] https://github.com/mapbox/mapbox-maps-ios-internal/issues/863

## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

Fixes: https://github.com/mapbox/mapbox-maps-ios-internal/issues/1174
Fixes: https://github.com/mapbox/mapbox-maps-ios-internal/issues/863

### CODE REVIEW

**Comment by macdrevx** [comment]
Please update the swift-tools-version in Package.swift

**Comment by macdrevx** [comment]
I think I'll also need to update depsvalidator b/c of a SPM format change

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,19 +42,19 @@ workflows:
     jobs:
       - swiftlint
       - depsvalidator:
-          xcode: "12.5.1"
+          xcode: "13.0.0"
```
Comment (macdrevx) [inline .circleci/config.yml:45]:
```suggestion
          xcode: "13.3.0"
```

File: scripts/depsvalidator/Sources/DepsValidatorLibrary/Models/Manifests/Package.swift [hunk 2]
```diff
@@ -74,6 +74,24 @@ struct Package: Decodable, SemanticValueProviding {
...
+    init(from decoder: Decoder) throws {
+        enum SourceControlNestedKeys: String, CodingKey {
+            case sourceControl
+        }
+        enum CodingKeys: String, CodingKey {
+            case name = "nameForTargetDependencyResolutionOnly"
+            case requirement
+        }
+        let container = try decoder.container(keyedBy: SourceControlNestedKeys.self)
+        var sourceControls = try container.nestedUnkeyedContainer(forKey: .sourceControl)
```
Comment (macdrevx) [inline scripts/depsvalidator/Sources/DepsValidatorLibrary/Models/Manifests/Package.swift:87]:
```suggestion
        let sourceControls = try container.nestedUnkeyedContainer(forKey: .sourceControl)
```

Can this be a let?

**Review by macdrevx — APPROVED** [review]

File: scripts/depsvalidator/Sources/DepsValidatorLibrary/Models/Manifests/Package.swift (hunk 2 above)
Comment (OdNairy) [reply scripts/depsvalidator/Sources/DepsValidatorLibrary/Models/Manifests/Package.swift:87]: It was the first time I worked with `UnkeyedDecodingContainer`. It happened that this protocol doesn't support `for-in` syntax and uses a weird one. 
It has an internal [`currentIndex`](https://developer.apple.com/documentation/swift/unkeyeddecodingcontainer/2921359-currentindex) property and [`count`](https://developer.apple.com/documentation/swift/unkeyeddecodingcontainer/2892719-count). Every time we call `decode…` or `…Container()` methods, we are incrementing the inner index 🤯.
All this methods are mutable, so it cannot be `let`.

**Pull request merged**